packs/softlayer/actions/lib/softlayer.py

```diff
@@ -0,0 +1,36 @@
+from libcloud.compute.providers import Provider
+from libcloud.compute.providers import get_driver
+from st2actions.runners.pythonrunner import Action
+
+
+class SoftlayerBaseAction(Action):
+    def __init__(self, config):
+        super(SoftlayerBaseAction, self).__init__(config=config)
+
+    def _get_driver(self):
+        cls = get_driver(Provider.SOFTLAYER)
+        return cls(self.config['username'], self.config['api_key'])
+
+    def _destroy_instance(self, **kwargs):
+        driver = self._get_driver()
+        nodes = [node for node in driver.list_nodes() if node.extra['hostname'] == kwargs['name']]
+        if len(nodes) == 1:
+            self.logger.info("Destroying node {} with remote ID {}".format(kwargs['name'], nodes[0].id))
+            return driver.destroy_node(nodes[0])
+        else:
+            self.logger.error("Node with name {} not found".format(kwargs['name']))
+            exit(1)
+
+    def _run_action(self, **kwargs):
```

Comment:
Please explicitly declare all the arguments here - it makes it easier to see what's going on.